### ISSUE TITLE
Both loss and accuracy increase when using ADAM on MNIST with Lenet (tensorflow)

### ISSUE THREAD

**Issue by CorentinJ** [NVIDIA/DIGITS#2119]
If you have MNIST, training the default Lenet model (using tensorflow as framework) using default training parameters with the exception of the solver set to ADAM, the accuracy will steadily increase but so will the training and validation loss. This will not happen if the Caffe version of Lenet is used or if SGD is the solver.

![image](https://user-images.githubusercontent.com/12038136/47016643-9c059300-d150-11e8-8237-c760d377f37a.png)
